src/scripts/main.js

```diff
@@ -1,3 +1,261 @@
+modifyElement(ageLabel, ageInput, 'Age', 'number');
+modifyElement(salaryLabel, salaryInput, 'Salary', 'number');
+
+formContainer.classList.add('new-employee-form');
+
+officeLabel.innerHTML = 'Office:';
+officeLabel.setAttribute('data-qa', 'office');
+officeLabel.appendChild(officeChoose);
+officeChoose.required = true;
+formButton.innerText = 'Save to table';
+
+cityOptions.forEach((el) => {
+  const option = document.createElement('option');
+
+  option.value = el;
+  option.innerText = el;
+  officeChoose.appendChild(option);
+});
+
+formContainer.appendChild(nameLabel);
+formContainer.appendChild(posiitionLabel);
+formContainer.appendChild(officeLabel);
+formContainer.appendChild(ageLabel);
+formContainer.appendChild(salaryLabel);
+formContainer.appendChild(formButton);
+
+pageBody.appendChild(formContainer);
+
+const pushNotification = (title, description, type) => {
+  const container = document.createElement('div');
+  const containerTitle = document.createElement('h2');
+  const containerDescr = document.createElement('p');
+
+  container.classList.add('notification');
+  container.setAttribute('data-qa', 'notification');
+  containerTitle.classList.add('title');
+
+  if (type === 'success' || type === 'error') {
+    container.classList.add('notification', type);
+  }
+  containerTitle.classList.add('title');
+  containerTitle.innerText = title;
+  containerDescr.innerText = description;
+
+  container.appendChild(containerTitle);
+  container.appendChild(containerDescr);
+  pageBody.appendChild(container);
+
+  setTimeout(() => {
+    container.style.display = 'none';
+  }, 3000);
+};
+
+formButton.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  if (
+    nameInput.value === '' ||
+    posiitionInput.value === '' ||
+    ageInput.value === '' ||
+    ageInput.value < 18 ||
+    ageInput.value > 100 ||
+    salaryInput.value === ''
+  ) {
+    return pushNotification(
+      'Error',
+      'Please, fix all the problems in form',
+      'error',
+    );
+  }
+
+  const newEmployee = document.createElement('tr');
+
+  for (let cell = 0; cell < inputs.length; cell++) {
+    const newCell = document.createElement('td');
+
+    if (cell === 4) {
+      const inputNumber = parseInt(inputs[cell].value);
+
+      newCell.innerHTML = `$${inputNumber.toLocaleString('en-US')}`;
+    } else {
+      newCell.innerHTML = inputs[cell].value;
+    }
+
+    newEmployee.appendChild(newCell);
+  }
+
+  inputs.forEach((input, i) => {
+    if (i !== 2) {
+      input.value = '';
+    }
+  });
+
+  tableBody.appendChild(newEmployee);
+
+  return pushNotification('Done', 'New employee was added to list', 'success');
+});
+
+tableBody.addEventListener('dblclick', (e) => {
+  const editedCell = e.target.closest('td');
+  const cellRow = e.target.closest('tr');
+  const indexOfCell = [...cellRow.children].indexOf(editedCell);
+  const cellInitialtName = editedCell.innerText;
+
+  if (editedCell.firstChild.tagName !== 'FORM') {
```

Comment:
Checking for a 'FORM' tag name directly is brittle. If the structure changes, this code will break. Consider using a more robust method to determine if the cell is in edit mode.